src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -546,7 +552,7 @@ private void CompileSynthesizedMethods(PrivateImplementationDetails privateImplC
             Debug.Assert(_moduleBeingBuiltOpt != null);
 
             var compilationState = new TypeCompilationState(null, _compilation, _moduleBeingBuiltOpt);
-            foreach (MethodSymbol method in privateImplClass.GetMethods(new EmitContext(_moduleBeingBuiltOpt, null, diagnostics)))
+            foreach (MethodSymbol method in privateImplClass.GetMethods(new EmitContext(_moduleBeingBuiltOpt, null, diagnostics, isRefAssembly: false)))
```

Comment:
I suspect this is wrong, but I'm not sure. I'll add some tests with private implementation details.